### PR TITLE
Upgrade to ocamlformat 0.26.0

### PR DESCRIPTION
**Pull request by Julow**
The changed formatting of `@@ let ...` (any operator) cause huge diffs in the command descriptions. The changed indentation of `if` and `let` in parentheses also has a big impact.  

The aim of this commit is to gather feedback.

Changelog can be found here: https://github.com/ocaml-ppx/ocamlformat/blob/main/CHANGES.md

### CODE REVIEW

**Comment by emillon** [comment]
Once 3.9.0 I'll upgrade to 0.25.0 first so we can see what's going on more precisely (I'll also enable `.git-blame-ignore-revs`).

File: src/dune_rules/dune_env.ml
Comment (rgrinberg) [inline src/dune_rules/dune_env.ml:236]: IMO the old code looked better here.

Comment (Julow) [reply src/dune_rules/dune_env.ml:236]: What do you think of the similar change in cmd code, where the let-expression is much bigger and the operators are not chained ?

**Comment by Julow** [comment]
> Once 3.9.0 is out I'll upgrade to 0.25.0 first so we can see what's going on more precisely (I'll also enable .git-blame-ignore-revs).

Let me know and I'll rebase this.

File: src/dune_rules/dune_env.ml
Comment (rgrinberg) [reply src/dune_rules/dune_env.ml:236]: The old version is better there too. The additional line only pays off when the let expression has some ugly wrapping due to the lack of horizontal space. There was no such issue there.

Comment (Alizter) [reply src/dune_rules/dune_env.ml:236]: Can ocamlformat pick the one that causes the least wrapping? (at least under the hood)

Comment (Julow) [reply src/dune_rules/dune_env.ml:236]: Not really, OCamlformat cannot compare two possible formatting. Only the input can be considered (though, multiple iterations on the whole code can be made).
But it can format differently whether there would be no wrapping at all and it can detect chained operators.

Such formatting could be possible:

```
<|> let+ p = Profile.decode in Profile p
<|> ...
```

falling back to this as soon as it doesn't fit on the line:

```
<|>
let+ p = Profile.decode in
(* longer code *)
Profile p
<|> ...
```

I'm mentioning chained operators because to me, the cmd code looks better with less indentation.
To me, that indentation *can* cause ugly wrapping means that it *does*.

File: bin/cache.ml
Comment (Julow) [inline bin/cache.ml:52]: The smaller indentation saved one line that was lost to wrapping here.

File: src/dune_rules/dune_project.ml
Comment (Julow) [inline src/dune_rules/dune_project.ml:918]: Same here.

File: src/dune_rules/dune_env.ml
Comment (Julow) [reply src/dune_rules/dune_env.ml:236]: By looking at the diff again, I find that the reduced indentation do save lines by removing some wrapping. I also do not find a lot of chained operators so my previous suggestion is not valid.

Though, it should be possible have this formatting, as long as it fits on one line:
```ocaml
...
<|> let+ p = Profile.decode in Profile p
```

Should I add this before the next release ?

Comment (emillon) [reply src/dune_rules/dune_env.ml:236]: not a huge fan of saving lines at all cost, I think that the old version was better.

Comment (Julow) [reply src/dune_rules/dune_env.ml:236]: I went back and found the reason for this change: https://github.com/ocaml-ppx/ocamlformat/pull/2315
It's a consistency change, in the previous version, the line was broken on regular lets but not letops:
```ocaml
let _ =
  Cmd.v info
  @@ let+ trimmed_size = () in
     ()

let _ =
  ()
  <|> let+ p = Profile.decode in
      Profile p

let _ =
  Cmd.v info
  @@
  let trimmed_size = () in
  ()

let _ =
  ()
  <|>
  let p = Profile.decode in
  Profile p
```

The other form exists in the code: `bin/describe/describe_workspace.ml` line 378.

This diff can go away by keeping the line break from the source code, whether it's a let or a letop. (https://github.com/ocaml-ppx/ocamlformat/pull/2396)
We try to add as few of these as possible but the regression is large enough to justify it.

I'll merge this patch and update this preview PR after the weekend.

**Comment by Julow** [comment]
I rebased and re-ran the preview with the `@@ let+` fix.

File: .git-blame-ignore-revs
Comment (emillon) [inline .git-blame-ignore-revs:4]: No need to do that in this PR since this will change

File: .ocamlformat
Comment (emillon) [inline .ocamlformat:1]:
```suggestion
version=0.26.0
```

File: bin/common.mli
Comment (emillon) [inline bin/common.mli:61]: I think we meant this to be a list. I mean, the output should include a literal `[`. Is this what this does?

Comment (Julow) [reply bin/common.mli:61]: Yes, balanced `[]` don't need escaping.

File: .git-blame-ignore-revs
Comment (Julow) [reply .git-blame-ignore-revs:4]: Right!

I'll make sure to remove that and fix the version line before I mark the PR as ready.

Comment (Julow) [reply .git-blame-ignore-revs:4]: Though, do you think this PR could be merged unsquashed ?

Comment (emillon) [reply .git-blame-ignore-revs:4]: for now I prefer doing this in 2 steps

Comment (Julow) [reply .git-blame-ignore-revs:4]: Ok!

**Review by emillon — APPROVED** [review]
In my opinion, these are all improvements. Let's wait for @rgrinberg's opinion as well.

**Comment by Julow** [comment]
Thanks :) Unless a problem is found, I intent to release today.

**Comment by Julow** [comment]
OCamlformat 0.26.0 has been released! This PR can be merged or closed if it's a bit too early to upgrade.

**Review by rgrinberg — APPROVED** [review]

**Comment by emillon** [comment]
I rebased this; this can land once #8311 is merged.

File: src/dune_rules/install_rules.ml
Comment (Alizter) [inline src/dune_rules/install_rules.ml:988]: Indentation here is kind of funny. Is this expected @Julow?

**Review by Alizter — APPROVED** [review]
LGTM, just a non-blocking comment.

**Pull request merged**
